src/crimson/osd/recovery_backend.cc

```diff
@@ -104,9 +104,7 @@ seastar::future<> RecoveryBackend::handle_backfill_progress(
     t);
   return shard_services.get_store().do_transaction(
     pg.get_collection_ref(), std::move(t)
-  ).handle_exception([] (auto) {
-    ceph_assert("this transaction shall not fail" == nullptr);
-  });
+  ).or_terminate();
```

Comment:
I think this might lead to loss of critical error information.

Reply:
is a backtrace good enough if it points to this line of code? i will revert this change anyway after Radek reviews this PR.

Reply:
Oh, yeah, that should be equally effective:)